packages/client/lib/miner/miner.ts

```diff
@@ -0,0 +1,217 @@
+      // will not have been initialized yet in the execution vm
+      // since the following line won't be reached:
+      // https://github.com/ethereumjs/ethereumjs-monorepo/blob/c008e8eb76f520df83eb47c769e3a006bc24124f/packages/client/lib/sync/execution/vmexecution.ts#L100
+      // So we will do it here:
+      await vmCopy.stateManager.generateCanonicalGenesis()
+    } else {
+      // Set the state root to ensure the resulting state
+      // is based on the parent block's state
+      await vmCopy.stateManager.setStateRoot(parentBlock.header.stateRoot)
+    }
+
+    let difficulty
+    if (this.config.chainCommon.consensusType() === ConsensusType.ProofOfAuthority) {
+      // Determine if signer is INTURN (2) or NOTURN (1)
+      const inTurn = vmCopy.blockchain.cliqueSignerInTurn(
+        signerAddress,
+        parentBlock.header.number.addn(1)
+      )
+      difficulty = inTurn ? 2 : 1
+    }
+
+    const blockBuilder = await vmCopy.buildBlock({
+      parentBlock,
+      headerData: {
+        difficulty,
+        gasLimit,
+      },
+      blockOpts: {
+        cliqueSigner: signerPrivKey,
+      },
+    })
+
+    let baseFee
+    if (this.config.chainCommon.isActivatedEIP(1559)) {
+      // Exclude txs below base fee
+      baseFee = parentBlock.header.calcNextBaseFee()
+    }
+    const txs = this.synchronizer.txPool.txsByPriceAndNonce(baseFee)
+    this.config.logger.info(
+      `Miner: Assembling block from ${txs.length} eligible txs ${
+        baseFee ? `(baseFee: ${baseFee.toNumber()})` : ''
+      }`
+    )
+    let index = 0
+    let blockFull = false
+    while (index < txs.length && !blockFull && !interrupt) {
+      try {
+        await blockBuilder.addTransaction(txs[index])
+      } catch (error) {
+        if (error.message === 'tx has a higher gas limit than the remaining gas in the block') {
+          if (blockBuilder.gasUsed.gt(gasLimit.subn(21000))) {
+            // If block has less than 21000 gas remaining, consider it full
+            blockFull = true
+            this.config.logger.debug(
+              `Miner: Assembled block full (gasLeft: ${gasLimit
+                .sub(blockBuilder.gasUsed)
+                .toNumber()})`
+            )
+          }
+        } else {
+          // If there is an error adding a tx, it will be skipped
+          const hash = '0x' + txs[index].hash().toString('hex')
+          this.config.logger.error(
+            `Skipping tx ${hash}, error encountered when trying to add tx:\n${error}`
+          )
+        }
+      }
+      index++
+    }
+    if (interrupt) return
+    // Build block, sealing it
+    const block = await blockBuilder.build()
+    this.config.logger.info(`Miner: Sealed block with ${block.transactions.length} txs`)
+    this.assembling = false
+    if (interrupt) return
+    // Put block in blockchain and remove included txs from TxPool
+    await this.synchronizer.handleNewBlock(block)
```

Comment:
Does making this call versus just `putBlock` assume that `handleNewBlock` will appropriately announce the new block to peers?  I think it makes sense that it would but just want to confirm as I'm starting to think about my next PR handling the rebroadcast mechanism for the `NEW_BLOCK` message.

Reply:
that makes sense to me!